### ISSUE TITLE
Freezes on exit, both SGP and WeatherStation

### ISSUE THREAD

**Issue by laperuz26** [manojkoushik/Arduino_ObservingConditions#4]
I'm putting together a simple weather station, mostly as a safety monitor for potential rain for my ROR observatory. Installed software on Arduino Uno and on my PC. Seems like all is reasonably well, except for one thing. I can't properly shut any program down. The Arduino weather station hangs on exit. SGP hangs on disconnecting "Arduino environmental device". Tried different settings, no difference.

Any ideas??

**Comment by Bombahook1** [comment]
Seems like this has been abandoned. You are a lot further along that I am. I'm still getting buggy errors trying to verify the code before uploading it to the Uno. Did you see the thread on Cloudy Nights?

**Comment by laperuz26** [comment]
Yes, I saw that thread and posted there. You're right,  it's not very active neither here, nor on CN. I will give the project another try in the Spring. Basically my hardware is all set. Hopefully I could make software to work. Will post here my updates/progress.

**Comment by Bombahook1** [comment]
I'm not as technically savvy as some of the people that create these things and it's hard to get up to speed. I understand using the programing environment, getting the libraries and uploading the code, but don't know how to troubleshoot when things don't work as planned. I was also confused about the humidity sensor in an enclosed box. Not sure how it's supposed to measure the humidity in a weather tight enclosure. 

The SparkFun store webpage has some good info, I'm going to try and follow some of their guides.

**Comment by manojkoushik** [comment]
hey all, sorry for the long delay in responding. November was tough with Covid and then i completely forgot to check here. 

@laperuz26 @Bombahook1 are you still noticing this error? any screenshots etc. you can provide? I don't have a weatherstation atm, but can definitely investigate.

**Comment by manojkoushik** [comment]
My suspicion is that I was sloppy in my cleanup on exit which is leaving something hanging.

**Comment by laperuz26** [comment]
I stopped working on the project. If manojkoushik is still active I'll resume and post here. Thanks!

**Comment by laperuz26** [comment]
I got another Arduino Board, Weather Shield, installed all the software and the libraries. Alas, it is still not working. Crashes NINA and ASCOM ObservingConditions Hub Tester. Basically it fails to connect properly and crashes on disconnect. Please let us know if you're still active. I think overall it's a great idea. I wish we could implement it. Thank you!

**Comment by manojkoushik** [comment]
Hey all, someone sent me a crash log which seemed to indicate a problem with the doubleparse() in the ascom driver. I.e. the ascom driver tried to read something from the arduino and got junk and crashed while trying to parse the text string into a double float. 

Do you know if this happens even when you deselect the "Generate Boltwood File" option in settings?

**Comment by manojkoushik** [comment]
Also it looks like it freezes ascom when trying to connect even initially. 

Could be because I have to do something to make it compatible with the latest ascom version. Will check.

**Comment by laperuz26** [comment]
Hi!
A few "dumb" questions:
1. Do I need to connect all (or maybe just MLX90614) to get a proper response from ASCOM driver?
2. Do I need to provide 12V to Arduino? It seems to make no difference weather or not I plug adapter in, but just wondering.
3. On Weather Shield, does a blue light have to blink in 15 sec (or other) interval set in WeatherStation?
4. When programming the Arduino, it seems there's a difference which of the IDE version being used - like various errors or no errors. Which one is better suited?
5. The Tetton program, is it necessary? I remember seeing a folder installed initially, then when I reinstalled everything - it wasn't there anymore.
Thank you for helping us sort this out. I still hope to make this work!

**Comment by pangalanopoulos** [comment]
It seems that this problem is still not fixed. Is this project still active?

**Comment by manojkoushik** [comment]
I just have not had time guys. I promise to spend some time this weekend
looking through the code. :(

**Comment by cloud4sure** [comment]
Mine hangs but also doesnt get any sensor readings when using the exe provided in the repo.
